packages/x-tree-view/src/internals/plugins/useTreeViewSelection/useTreeViewSelection.itemPlugin.ts

```diff
@@ -72,6 +70,17 @@ export const useTreeViewSelectionItemPlugin: TreeViewItemPlugin = ({ props }) =>
     store,
     selection: { disableSelection, checkboxSelection, selectionPropagation },
   } = useTreeViewContext<[UseTreeViewItemsSignature, UseTreeViewSelectionSignature]>();
+
+  const checkboxStatus = useSelector(
```

Comment:
I was accessing the state without `useSelector` so it was not triggering a re-render